gpu/src/Sampler.zig

```diff
@@ -36,7 +36,7 @@ pub const BindingType = enum(u32) {
 
 pub const BindingLayout = extern struct {
     reserved: ?*anyopaque = null,
-    type: BindingType,
+    type: BindingType = .none,
```

Comment:
Similarly, should default to `.filtering`

<img width="448" alt="image" src="https://user-images.githubusercontent.com/3173176/162637549-8317c3b4-91c8-4103-9cde-083a8ccfba6b.png">

https://www.w3.org/TR/webgpu/#dictdef-gpusamplerbindinglayout